### ISSUE TITLE
[$450] Implement admin piece in TC plugin

### ISSUE THREAD

**Issue by jmgasper** [topcoder-platform/forums-plugins#15]
### Required functionality
I'd like to extend the current plugin to add a new tab and / or new options to an existing tab, on the dashboard site with the required functionality for admins.  

- Admin have the ability to identify all messages from a user
- Platform must support the ability to export data on a per-user basis
- Must log who posted a message, date, time, and contents if text (seems like this is already done by default - I'm getting clarification)

![Screen Shot 2020-05-06 at 09 41 26](https://user-images.githubusercontent.com/21790/81126142-c2cb2a80-8f7d-11ea-8fd8-4e6bf7d11606.png)

### Output format

The output format can be pretty raw.  I don't anticipate needing to convert from what we get from the REST methods - the goal here is just to provide the data, but there are no format requirements.  JSON is fine.

### Discussion

As discussed in #14 , it seems like this should be doable.

**Comment by jmgasper** [comment]
Contest https://www.topcoder.com/challenges/30124703 has been created for this ticket.<br/><br/>```This is an automated message for ghostar via Topcoder X```

**Comment by jmgasper** [comment]
Contest https://www.topcoder.com/challenges/30124703 has been updated - it has been assigned to obog.<br/><br/>```This is an automated message for ghostar via Topcoder X```

**Comment by atelomycterus** [comment]
@jmgasper  1. What about images and attachments uploaded by the user to the forum? Should we export these details in a json file? Should  we export images/attachments as a zip archive?
Typically, users participating in challenges do not upload personal data.
If we plan to replace the existing forum, then perhaps there willbe user support category/taas groups, where users must pass verification, for example, some customers require passport verification or so on.
Images/Attachments may also contain personal data. 

2. One json file will be created. The structure will be like this:   

```
{
  "user": {}
  "discussions": [],
  "comments": [],
  "drafts": [],
  "conversations": [],
   .....
}
```

**Comment by jmgasper** [comment]
@atelomycterus - I'll ask about attachments and what the requirements are there.  The JSON format looks great, thanks!

**Comment by atelomycterus** [comment]
@jmgasper   a couple of questions:
 
1. Vanilla easily allows to expand dashboard screens without coping-pasting views.
I added an export button at 'Manage Users'. 
Let me know if you are ok with UI.  

![image](https://user-images.githubusercontent.com/10196632/81472468-ed78e580-9200-11ea-8335-95257bbf878a.png)

2. The user with permissions ('Garden.Users.Add','Garden.Users.Edit','Garden.Users.Delete') only has access to the 'Manage Users'. We can create our own permission to show/hide the export button.
Let me know if it's required. 

3. Do we need a date range filter to export data? It might be implemented as a popup.  
Now all data is uploaded per an user. Let me know if it's required. 

P.S. The pull request - https://github.com/topcoder-platform/forums-plugins/pull/16
Export data (user, discussions, comments, conversations, messages, drafts, IPs) was implemented.
Example jsons:
[export-reports.zip](https://github.com/topcoder-platform/forums-plugins/files/4603994/export-reports.zip)


Open questions:
- export images/attachments
- 'forgotten' rights

**Comment by jmgasper** [comment]
@atelomycterus - Sorry for the delayed response here:

1.  UI Looks fine, thanks
2.  No, this will only be used by admins, so that list of permissions is fine.
3.  No, I don't think we need a date range yet.  If we do, we'll create a new ticket for that, thanks.

For now, don't worry about exporting images / attachments or the forgotten rights, thanks.

**Comment by atelomycterus** [comment]
@jmgasper Thanks for details. Please apply PR - https://github.com/topcoder-platform/forums-plugins/pull/16.

**Comment by jmgasper** [comment]
Looks good.  I'll get feedback from Topcoder to see if this meets their needs.

**Comment by jmgasper** [comment]
Will close this once I get final clarification on a billing account change.  There was a problem here that was still being resolved, sorry.

**Comment by jmgasper** [comment]
@atelomycterus - Sorry for the delay, still waiting on further clarifications from Topcoder RE: billing account.

**Comment by jmgasper** [comment]
Payment failed: [500]: Failed to get billing detail for the project.<br/><br/>```This is an automated message for ghostar via Topcoder X```

**Comment by jmgasper** [comment]
This has been paid

**Comment by jmgasper** [comment]
This ticket was not processed for payment. If you would like to process it for payment, please reopen it, add the ```tcx_FixAccepted``` label, and then close it again<br/><br/>```This is an automated message for ghostar via Topcoder X```
